### PR TITLE
Added AdvancedSettingsMixin

### PR DESCRIPTION
**Pull request by meisnate12**
## Description

I moved preferences, editAdvanced, and defaultAdvanced methods from video.Show in a new mixin called AdvancedSettingsMixin and added the mixin to video.Show, video.Movie, and audio.Artist

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update (I'm not sure if the docs would need an update or not)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [x] I have added tests when applicable (I could use some help writing proper tests for this)

### CODE REVIEW

**Comment by Hellowlol** [comment]
Hi! Remember to test your changes. This will fail. Check the flake output 🙂

File: plexapi/audio.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def sync(self, bitrate, client=None, clientId=None, limit=None, title=None):
 
 @utils.registerPlexObject
 class Artist(Audio, ArtMixin, PosterMixin, SplitMergeMixin, UnmatchMatchMixin,
-        CollectionMixin, CountryMixin, GenreMixin, MoodMixin, SimilarArtistMixin, StyleMixin):
+        CollectionMixin, CountryMixin, GenreMixin, MoodMixin, SimilarArtistMixin, StyleMixin, AdvancedSettingsMixin):
```
Comment (JonnyWong16) [inline plexapi/audio.py:118]: Let's keep the order of the Mixins. Put `AdvancedSettingsMixin` before `ArtMixin`.

File: plexapi/video.py [hunk 2]
```diff
@@ -248,7 +248,7 @@ def sync(self, videoQuality, client=None, clientId=None, limit=None, unwatched=F
 
 
 @utils.registerPlexObject
-class Movie(Video, Playable, ArtMixin, PosterMixin, SplitMergeMixin, UnmatchMatchMixin,
+class Movie(Video, Playable, ArtMixin, PosterMixin, SplitMergeMixin, UnmatchMatchMixin, AdvancedSettingsMixin,
```
Comment (JonnyWong16) [inline plexapi/video.py:251]: Same `AdvancedSettingsMixin` before `ArtMixin`.

File: plexapi/video.py [hunk 3]
```diff
@@ -382,7 +382,7 @@ def download(self, savepath=None, keep_original_name=False, **kwargs):
 
 @utils.registerPlexObject
 class Show(Video, ArtMixin, BannerMixin, PosterMixin, SplitMergeMixin, UnmatchMatchMixin,
-        CollectionMixin, GenreMixin, LabelMixin):
+        CollectionMixin, GenreMixin, LabelMixin, AdvancedSettingsMixin):
```
Comment (JonnyWong16) [inline plexapi/video.py:385]: Same `AdvancedSettingsMixin` before `ArtMixin`.

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]

**Review by JonnyWong16 — APPROVED** [review]

**Comment by JonnyWong16** [comment]
Tests added. Ready to merge.

**Pull request merged**